kurtosis_package/dockerfiles/contract_deployer.Dockerfile

```diff
@@ -1,46 +1,40 @@
-FROM debian:bookworm-slim AS foundry
+FROM debian:bookworm-slim as foundry
 
-# Install curl & git
-RUN apt update -y && \
-    apt upgrade -y && \
+# Install required tools
+RUN apt update && \
     apt install -y curl git && \
     apt clean
 
-# Install foundry
-RUN curl -L https://foundry.paradigm.xyz | bash
-ENV PATH="/root/.foundry/bin:${PATH}"
+# Install Foundry
+RUN curl -L https://foundry.paradigm.xyz | bash && \
+    /root/.foundry/bin/foundryup
 
-RUN foundryup
+ENV PATH="/root/.foundry/bin:$PATH"
 
-WORKDIR /app/
+# Create /app and make it writable
+RUN mkdir -p /app && chmod -R 777 /app
 
+# Use /app as working directory
+WORKDIR /app
 
 # The repository's URL.
-ARG CONTRACTS_REPO
+ARG CONTRACTS_REPO=""
 # The commit hash, tag, or branch to checkout from the repo.
-ARG CONTRACTS_REF
+ARG CONTRACTS_REF=""
 # Path to the contracts directory within the repository.
 # It must contain a foundry.toml file.
 ARG CONTRACTS_PATH="."
 
-FROM foundry AS contract_deployer
-
-WORKDIR /
-
-ARG CONTRACTS_REPO
-ARG CONTRACTS_REF
-ARG CONTRACTS_PATH
-
-RUN git init app
-
-WORKDIR /app
-
-RUN git remote add origin ${CONTRACTS_REPO} && \
-    git fetch --depth 1 origin ${CONTRACTS_REF} && \
+# Only clone if CONTRACTS_REPO is not empty
+RUN if [ -n "$CONTRACTS_REPO" ]; then \
```

Comment:
Why would users use this without `CONTRACTS_REPO` set?

Reply:
For cases where local repo is being used
https://github.com/Layr-Labs/avs-devnet/blob/94d76376293d6fc3f7ae1ac4f81c983bf6446501/kurtosis_package/deployers/utils.star#L31